ddtrace/_trace/tracer.py

```diff
@@ -232,7 +233,18 @@ def __init__(
         self.context_provider = context_provider or DefaultContextProvider()
         # _user_sampler is the backup in case we need to revert from remote config to local
         self._user_sampler: Optional[BaseSampler] = DatadogSampler()
+        self._asm_enabled = asm_config._asm_enabled
         self._sampler: BaseSampler = DatadogSampler()
+        self._apm_opt_out = self._asm_enabled and not self.enabled
+        if self._apm_opt_out:
+            # If ASM is enabled but tracing is disabled,
+            # we need to set the rate limiting to 1 trace per minute
+            # for the backend to consider the service as alive.
+            from ddtrace.internal.rate_limiter import RateLimiter
+
+            self._sampler.limiter = RateLimiter(rate_limit=1, time_window=60e9)  # 1 trace per minute
```

Comment:
Technically we can't rely on the _sampler always having a rate limiter in the event that customers set their own custom sampler (deprecated and is going to be removed in next major release). So we may want to try except around this and log something.

Reply:
Changing that in the setter of the sampler instead, to deactivate that feature if Standalone ASM is enabled